test/iterable_test.dart

```diff
@@ -951,5 +951,17 @@ void main() {
         expect(accessCount, 3);
       });
     });
+
+    test('.startsWith()', () {
+      expect([].startsWith([]), true);
+      expect([1, 2, 3].startsWith([]), true);
+      expect([1, 2, 3].startsWith([1]), true);
+      expect([1, 2, 3].startsWith([1, 2]), true);
+      expect([1, 2, 3].startsWith([1, 2, 3]), true);
+      expect([1, 2, 3].startsWith([1, 2, 3, 4]), false);
+      expect([1, 2, 3].startsWith([2, 3, 4]), false);
+      expect([1, 2, 3].startsWith([2, 3]), false);
+      expect([1, 2, 3].startsWith([2]), false);
```

Comment:
please add a test with `null` values. There might be some tricky corner cases.

Also add a test where `this` is shorter than `otherIterable`

Reply:
will do the `null` test.

the test in line 961 contains a shorter list for `this` than for the `otherIterable`. or do you mean something else?

Reply:
passing `null` values to the method is not possible as the method does not accept `nullable` values. a `!` can be appended but this throws an exception before the method is invoked.

```dart
const List<int>? nullList = null;
expect([1, 2, 3].startsWith(nullList!), null);
```
```bash
Null check operator used on a null value
main.<fn>.<fn>
✖ IterableX .startsWith()
Exited (1)
```

 `this` can be null with a `null-aware` operator. in case of a `?` the method does not even get invoked and `null` is directly returned. in case of a `!` an exception is thrown and the method does not get invoked either. I don't see the necessity to test this. 

```dart
const List<int>? nullList = null;
expect(nullList?.startsWith([1, 2, 3]), null);
expect(nullList!.startsWith([1, 2, 3]), null);
```
```bash
Null check operator used on a null value
main.<fn>.<fn>
✖ IterableX .startsWith()
Exited (1)
```

Reply:
@passsy please check my comments. if i missed something or similar, please let me know.

Reply:
Please add the following casees
- `expect([].startsWith([1, 2, 3]), isFalse);` and 
- `expect([null, 1, 2].startsWith([null, 1]), isFalse) `

Reply:
Oh, you mean `null` elements in the list and not the list itself. Sorry, I should have thought about this.

I don't see the reason why the method should return false if both have a `null` value at the same index. I would argue that the method should return true because both values are equal and it's in the user's responsibility to check for `null` elements. What do you think?

Reply:
My fault, of course it should return `true` (corrected it).

Reply:
added the tests.